### PR TITLE
Add more missing error checks in lib/*

### PR DESCRIPTION
**Pull request by awly**
Another batch of `errcheck` findings.
Most of these will just get logged for debugging.

Updates https://github.com/gravitational/teleport/issues/3551

### CODE REVIEW

File: lib/srv/authhandlers.go [hunk 1]
```diff
@@ -109,15 +109,17 @@ func (h *AuthHandlers) CheckPortForward(addr string, ctx *ServerContext) error {
...
 			events.PortForwardSuccess: false,
 			events.PortForwardErr:     systemErrorMessage,
 			events.EventLogin:         ctx.Identity.Login,
 			events.EventUser:          ctx.Identity.TeleportUser,
 			events.LocalAddr:          ctx.Conn.LocalAddr().String(),
 			events.RemoteAddr:         ctx.Conn.RemoteAddr().String(),
-		})
+		}); err != nil {
+			h.Warnf("Failed to emit port forward deny audit event: %v", err)
+		}
```
Comment (webvictim) [inline lib/srv/authhandlers.go:122]: nit: This seems quite long for the one-line `if err := blah(); err != nil` type syntax. I don't know enough about Go reviews to say whether it's something that should generally be avoided or not though.

File: lib/srv/exec_test.go [hunk 2]
```diff
@@ -282,21 +283,20 @@ func (s *ExecSuite) TestContinue(c *check.C) {
 	cmd, err := ConfigureCommand(ctx)
 	c.Assert(err, check.IsNil)
 
-	// Create a context that will be used to signal that execution is complete.
-	doneContext, doneCancel := context.WithCancel(context.Background())
+	// Create a channel that will be used to signal that execution is complete.
+	cmdDone := make(chan error, 1)
 
 	// Re-execute Teleport and run "ls". Signal over the context when execution
```
Comment (webvictim) [inline lib/srv/exec_test.go:289]: nit: Is this signalling over a channel rather than a context now?

**Review by webvictim — APPROVED** [review]

File: lib/srv/authhandlers.go (hunk 1 above)
Comment (awly) [reply lib/srv/authhandlers.go:122]: Logically it's one long line, yeah.
But practically it's easy enough to distinguish the different parts here. I've seen many such snippets, it's generally acceptable.

File: lib/srv/exec_test.go (hunk 2 above)
Comment (awly) [reply lib/srv/exec_test.go:289]: Yes, and also bubbles up the error from `cmd.Run()`.
It's unsafe to call `c.Assert` from goroutines.

**Pull request merged**